airflow/models/skipmixin.py

```diff
@@ -60,17 +61,24 @@ class SkipMixin(LoggingMixin):
     def _set_state_to_skipped(
         self,
         dag_run: DagRun | DagRunPydantic,
-        tasks: Iterable[Operator],
+        tasks: Iterable[str] | Iterable[tuple[str, int]],
         session: Session,
+        include_map_index: bool = False,
```

Comment:
The combination of these arguments is weird. If you don’t want to include map indexes, just don’t pass them? An `isinstance` check on `tasks` would be sufficient instead of a separate flag.

Reply:
fixed